cranelift/codegen/src/isa/aarch64/lower_inst.rs

```diff
@@ -49,6 +49,15 @@ pub(crate) fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
     match op {
         Opcode::Iconst | Opcode::Bconst | Opcode::Null => {
             let value = ctx.get_constant(insn).unwrap();
+            // Sign extend constant if necessary
+            let value = match ty.unwrap() {
+                I8 => (((value as i64) << 8) >> 8) as u64,
+                I16 => (((value as i64) << 16) >> 16) as u64,
+                I32 => (((value as i64) << 32) >> 32) as u64,
```

Comment:
Is it sign extention from small ints to `i64`, right? Perhaps I'm missing something, that it should look like this I guess:
```suggestion
                I8 => (((value as i64) << 56) >> 56) as u64,
                I16 => (((value as i64) << 48) >> 48) as u64,
                I32 => (((value as i64) << 32) >> 32) as u64,
```

Reply:
Indeed. Good catch! Thanks!